main.tf

```diff
@@ -355,6 +367,28 @@ module "atlantis_sg" {
   tags = merge(local.tags, var.atlantis_security_group_tags)
 }
 
+module "efs_sg" {
+  count = var.enable_ephemeral_storage ? 0 : 1
+  source = "terraform-aws-modules/security-group/aws"
+  version = "v4.8.0"
+
+  name = "${var.name}-efs"
+  vpc_id = local.vpc_id
+  description = "Security group allowing access to the EFS storage"
+
+  ingress_with_source_security_group_id = [
```

Comment:
There is a separate submodule for `nfs/efs` - https://github.com/terraform-aws-modules/terraform-aws-security-group/tree/master/modules/nfs ([usage example for submodules](https://github.com/terraform-aws-modules/terraform-aws-security-group/blob/master/examples/http/main.tf)).

It will make your code a few lines shorter.